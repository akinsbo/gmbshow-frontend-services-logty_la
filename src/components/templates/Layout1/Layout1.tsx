import { createStyles, makeStyles, Theme } from '@material-ui/core/styles'
// tslint:disable-next-line:import-name
import * as React from 'react'
import Header from '../../organisms/Header/Header'
// import VideoSearchBar from '../../molecules/VideoSearchBar'

// create a styles object using a theme. The createStyles function is
// needed to placate the TS compiler. 
// See https://gist.github.com/johnsogg/dde852d76aa7af92c1cedd0e594557db
const useStyles = makeStyles((theme: Theme) =>
    createStyles({
        root: {
            width: '100%',
        },
        grow: {
            flexGrow: 1,
        },
        menuButton: {
            marginRight: theme.spacing(2),
        },
        title: {
            display: 'none',
            [theme.breakpoints.up('sm')]: {
                display: 'block',
            },
        },
        input: {
            display: 'none',
        },
    })
)


// export interface HeaderProps extends WithStyles<typeof styles> { }
// Take children as props and wrap them
// const Layout1 = (props: { children: JSX.Element[] | JSX.Element | React.ReactNode}) => {
const Layout1 = (props: { children: React.ReactNode}) => {

    const classes = useStyles()

    return (
        <React.Fragment >
            <div className={classes.grow}>
                <Header />
                Loading Base ...
                {props.children}
            </div>
        </React.Fragment >
    )
}

export default Layout1
