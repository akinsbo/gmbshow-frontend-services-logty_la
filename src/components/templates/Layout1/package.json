{
    "name": "Layout1",
    "version": "1.0.0",
    "description": "Layout1 component",
    "main": "Layout1.tsx",
    "scripts": {
        "test": "jest"
    },
    "repository": {
        "type": "git",
        "url": "https://github.com/akinsbo/gmbshow-frontend-services-logty_la/src/components/atoms/Layout1"
    },
    "keywords": [
        "Base",
        "component",
        "template",
        "material"
    ],
    "author": "Olaolu Akinsete",
    "license": "ISC"
}