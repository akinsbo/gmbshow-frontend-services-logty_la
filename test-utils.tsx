import React from 'react'
import { render, queries } from '@testing-library/react'
import { ThemeProvider } from '@material-ui/styles'
import { createMuiTheme } from '@material-ui/core/styles'
import { I18nextProvider } from 'react-i18next'
import defaultStrings from './i18n/en-x-default'
import * as customQueries from './custom-queries'
import grey from '@material-ui/core/colors/grey'
import orange from '@material-ui/core/colors/orange'
import purple from '@material-ui/core/colors/purple'
import yellow from '@material-ui/core/colors/yellow'
// A theme with custom primary and secondary color.
// It's optional.
const theme = createMuiTheme({
    palette: {
        primary: {
            dark: purple[900],
            light: orange[500],
            main: grey[700]
        },
        secondary: {
            dark: yellow[700],
            light: purple[300],
            main: yellow[500]
        },
    }
})

const AllTheProviders = ({ children }: any) =>
    <ThemeProvider theme={theme} >
        <I18nextProvider i18n={defaultStrings} >
            {children}
        </I18nextProvider>
    </ThemeProvider >


const customRender = (ui: any, options = {}) =>
    render(ui, { wrapper: AllTheProviders, queries: { ...queries, ...customQueries }, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }