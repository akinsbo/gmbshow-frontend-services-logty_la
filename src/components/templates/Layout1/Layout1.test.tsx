import * as React from "react"
import { render, screen } from '../../../../test-utils';
import Layout1 from "./Layout1"


describe("Layout1", () => {
    it("renders correctly", () => {
        const container = render(
            <Layout1 ><div>Layout</div> </Layout1>
        )
        expect(container).toMatchSnapshot()
    })

    it('should contain a Header and a Footer', () => {
        // check header is present by checking for data-cy heaer or "Home, about, )"
        render(
            <Layout1 >
                <div>Layout</div>
            </Layout1>
        )
        expect(screen.getByText('Layout')).toBeTruthy()
    })
})

