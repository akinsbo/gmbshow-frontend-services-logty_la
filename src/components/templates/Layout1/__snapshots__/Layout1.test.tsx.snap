// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout1 renders correctly 1`] = `undefined`;
